cpp/src/strings/convert/convert_hex.cu

```diff
@@ -254,7 +254,7 @@ std::unique_ptr<column> is_hex(strings_column_view const& strings,
                       };
                       auto begin = d_str.begin() + (starts_with_0x(d_str) ? 2 : 0);
                       auto end   = d_str.end();
```

Comment:
```suggestion
                      auto begin = d_str.data() + (starts_with_0x(d_str) ? 2 : 0);
                      auto end   = begin + d_str.size_bytes();
```
This should fix the compile error here. 
Just an alternate implementation that is still valid because we expect the input data to be ASCII but it will still work for UTF-8 as well.